tests/wrapper/test_simulation_objects.py

```diff
@@ -28,3 +28,132 @@ def test_update_state(self, traffic_update):
         signal.state = Signal.State.GO
 
         assert signal.state == Signal.State.GO
+
+
+class TestTrain:
+    """Tests for the train object"""
+
+    @pytest.fixture
+    def train(self):
+        created_train = Train(
+            identifier="fake-sim-train",
+            train_type="fancy-ice",
+            timetable=[],
+            from_simulator=True,
+        )
+        created_train.update(
+            {
+                constants.VAR_POSITION: (
+                    100,
+                    100,
+                ),
+                constants.VAR_ROAD_ID: 123,
+                constants.VAR_ROUTE: "testing-route",
+                constants.VAR_SPEED: 10.2,
+                constants.VAR_MAXSPEED: 11,
+            }
+        )
+
+        return created_train
+
+    @pytest.fixture
+    def vehicle_route(self, monkeypatch):
+        def set_route_id(train_id: str, route_id: str):
+            assert train_id == "fake-sim-train"
+            assert route_id == "testing-route-beta"
+
+        monkeypatch.setattr(vehicle, "setRouteID", set_route_id)
+
+    @pytest.fixture
+    def max_speed(self, monkeypatch):
+        def set_max_speed(train_id: str, speed: float):
+            assert train_id == "fake-sim-train"
+            assert speed == 100
+
+        monkeypatch.setattr(vehicle, "setMaxSpeed", set_max_speed)
+
+    @pytest.fixture
+    def train_add(self, monkeypatch):
+        def add_train(identifier, route, train_type):
+            assert identifier == "fancy-rb-001"
+            assert route == "not-implemented"
+            assert train_type == "fancy-rb"
+
+        monkeypatch.setattr(vehicle, "add", add_train)
+
+    def test_track(self, train):
+        assert train.track == 123
+
+    def test_position(self, train):
+        assert train.position == (
+            100,
+            100,
+        )
+
+    def test_speed(self, train):
+        assert train.speed == 10.2
+
+    def test_route(self, train):
+        assert train.route == "testing-route"
+
+    def test_set_route(self, train, vehicle_route):
+        # pylint: disable=unused-argument
+        train.route = "testing-route-beta"
+        assert train.route == "testing-route-beta"
+
+    def test_max_speed(self, train):
+        assert train.max_speed == 11
+
+    def test_set_max_speed(self, train, max_speed):
+        # pylint: disable=unused-argument
+        train.max_speed = 100
+        assert train.max_speed == 100
+
+    def test_priority(self, train):
+        assert train.priority == 0
+
+    def test_set_priority(self, train):
+        train.priority = 10
+        assert train.priority == 10
```

Comment:
Some of these test look rather trivial, but I guess that is better than not having them.